src/react/databrowser/objects/details/__tests__/Metadata.test.tsx

```diff
@@ -32,76 +36,146 @@ describe('Metadata', () => {
-    const optionsComponents = firstItem.find('Option');
-    optionsComponents.forEach((optionComponent, index) => {
-      expect(optionComponent.find('.sc-select__option').first().text()).toBe(
-        optionLabels[index],
-      );
-    });
-    // select a key
-    const firstOption = optionsComponents.first();
-    firstOption.simulate('click');
-    // check dropdown menu disappear
-    items = component.find(Item);
-    firstItem = items.first();
-    expect(firstItem.find('Option')).toHaveLength(0);
-    // add a value to the key
-    firstItem.find('input.metadata-input-value').simulate('change', {
-      target: {
-        value: 'value1',
-      },
-    });
-    // click on add button
-    firstItem.find('button#addbtn0').simulate('click');
-    // check if new item row added
-    expect(component.find(Item)).toHaveLength(2);
-    // check if function when pressing save button is triggered
+    const { component: { container } }  = reduxRender(
+      <Metadata
+        objectMetadata={{
+          ...OBJECT_METADATA,
+          metadata: [
+            {
+              key: 'CacheControl',
+              value: 'no-cache',
+              type: METADATA_SYSTEM_TYPE,
+            },
+          ],
+        }}
+      />
+      , {
+        networkActivity: {
+          counter: 0,
+          messages: List.of(),
+        },
+        instances: {
+          selectedId: instanceId,
+        },
+        auth: {
+          config: { features: [] },
+          selectedAccount: { id: accountId },
+        },
+        oidc: {
+          user: {
+            access_token: ''
+          }
+        },
+        configuration: {
+          latest: {
+            endpoints: [],
+          },
+        },
+      });
+
+    expect(screen.getByRole('textbox', {name:''}  ).id).toBe('mdValue');
+    expect(screen.getByRole('textbox', {name:'select'}  )).toBeInTheDocument();
+    expect(container.getElementsByClassName('sc-select')[0].lastChild).toHaveValue('CacheControl');
+    expect(screen.getByRole('textbox', {name:''}  )).toHaveValue('no-cache');
+    expect(screen.getByRole('button', {name:'Remove'}  )).toBeDisabled();
+    expect(screen.getByRole('button', {name:'Add'}  )).not.toBeDisabled();
+    expect(screen.getByRole('button', {name:'Save'}  )).toBeDisabled();
+    userEvent.type(screen.getByRole('textbox', {name:''}  ), 'newValue');
+    userEvent.click(screen.getByRole('textbox', {name:'select'}  ));
+    userEvent.type(screen.getByRole('textbox', {name:'select'}  ), 'newKey');
+    expect(putObjectMetadataMock).toHaveBeenCalledTimes(0);
+    expect(screen.getByRole('button', {name:'Save'}  )).not.toBeDisabled();
+    fireEvent.click(screen.getByRole('button', {name:'Save'}  ));
     expect(putObjectMetadataMock).toHaveBeenCalledTimes(0);
-    component.find('button#metadata-button-save').simulate('click');
-    expect(putObjectMetadataMock).toHaveBeenCalledTimes(1);
   });
   it('remove button and add button should be disabled', () => {
```

Comment:
By default the SubButton and AddButton are disabled when the toggle  list versions is active. But this element is present in another component. Otherwise, the two buttons should be enabled.